scheduler/system_sched.go

```diff
@@ -127,6 +129,19 @@ func (s *SystemScheduler) process() (bool, error) {
 		s.logger.Printf("[DEBUG] sched: %#v: rolling update limit reached, next eval '%s' created", s.eval, s.nextEval.ID)
 	}
 
+	// If there are failed allocations, we need to create a blocked evaluation
+	// to place the failed allocations when resources become available.
+	if len(s.plan.FailedAllocs) != 0 && s.blocked == nil {
+		e := s.ctx.Eligibility()
+		classes := e.GetClasses()
+		s.blocked = s.eval.BlockedEval(classes, e.HasEscaped())
```

Comment:
There is the case in which you run a system job and there are no resources on a node that is eligible. In this case it should block till resources become available.